### DISCUSSION TITLE
[@types/lodash] Incorrect get() return value inference in v4.14.191

### DISCUSSION THREAD

**Discussion by jpveooys** [DefinitelyTyped/DefinitelyTyped#63791] [Issues with a @types package]
Hi,

I'm trying to update from v4.14.186 to v4.14.191 and hit into problems with incorrect return value types being inferred for `get()`.

Below is a simplified example:

```typescript
function getValue(keyA: 'one' | 'two', keyB: 'a' | 'b'): string {
  const tokens = {
    root: {
      one: {
        a: {
          value: '123',
        },
        b: {
          value: '123',
        },
      },
      two: {
        a: {
          value: '123',
        },
        b: {
          value: '123',
        },
      },
    },
  }

  // TS2322: Type 'undefined' is not assignable to type 'string'
  return get(tokens, `root[${keyA}][${keyB}].value`)
}
```

`undefined` is inferred when it should in fact be `string` in this case. You can run `getValue('one', 'a')` to verify that `'123'` is returned. 

This is a simplified example, the actual original example is more complex (i.e. there could be other similar problems as well).

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
@jpveooys does it work if you change it to:

```ts
get(tokens, `root.${keyA}.${keyB}.value`)
```

@atsikov it looks like they are using the list-style syntax for objects. Is it possible to support this?

**Reply by jpveooys** [reply]
> @jpveooys does it work if you change it to:
> 
> ```ts
> get(tokens, `root.${keyA}.${keyB}.value`)
> ```

It would, but it would break if a key contained a `.` for example. Using an array for the path would probably be the best option in terms of using `get()`.
